src/terminal/RenderBufferBuilder.cpp

```diff
@@ -449,6 +449,7 @@ ColumnCount RenderBufferBuilder<Cell>::renderUtf8Text(CellLocation screenPositio
     auto graphemeClusterSegmenter = unicode::utf8_grapheme_segmenter(text);
     for (u32string const& graphemeCluster: graphemeClusterSegmenter)
     {
+        screenPosition.column = ColumnOffset::cast_from(columnCountRendered);
         auto const gridPosition = terminal.viewport().translateScreenToGridCoordinate(screenPosition);
```

Comment:
```suggestion
        auto const gridPosition = terminal.viewport().translateScreenToGridCoordinate(screenPosition + ColumnOffsset::cast_from(columnCountRendered));
```

Instead of modifying `screenPosition`, I think this should do it, too (then the lines below should probably be undone).